qutebrowser/browser/hints.py

```diff
@@ -130,7 +135,23 @@ def _move_to_elem(self):
     def cleanup(self):
         """Clean up this element and hide it."""
         self.hide()
-        self.deleteLater()
+        self.setParent(None)
```

Comment:
After we put a limit on the size of the pool, I think this will LGTM. Ideally we would want to set this high enough so this will never be reached in typical browsing, but rather just a limit so if there's an extreme case it won't lead to many objects sitting around for no reason. I'm not sure what a good value is at the moment though...

Reply:
@jgkamat Alternatively, we can have a smart logic to free up the object pool when they're not used for a while. They're likely to take much less time than initialize and render the hint labels anyway.

Reply:
Depends on how much complexity that'd add, which is hard to tell without seeing the code :wink: Alternatively, I guess a limit like caching 500 objects (I get ~350 on a reddit post with the old layout) would work as well, just as a safety net. But it also depends on how much memory that'd use overall...